runtime/envelope/envelope.go

```diff
@@ -35,55 +35,11 @@ import (
 	"github.com/ServiceWeaver/weaver/runtime/metrics"
 	"github.com/ServiceWeaver/weaver/runtime/protos"
 	"github.com/ServiceWeaver/weaver/runtime/retry"
-	"go.opentelemetry.io/otel/sdk/trace"
 )
 
 // EnvelopeHandler implements the envelope side processing of messages
 // exchanged with the managed weavelet.
-type EnvelopeHandler interface {
-	// StartComponent starts the given component.
-	StartComponent(entry *protos.ComponentToStart) error
-
-	// StartColocationGroup starts the given colocation group.
-	StartColocationGroup(entry *protos.ColocationGroup) error
-
-	// RegisterReplica registers the given weavelet replica.
-	RegisterReplica(entry *protos.ReplicaToRegister) error
-
-	// ReportLoad reports the given weavelet load information.
-	ReportLoad(entry *protos.WeaveletLoadReport) error
-
-	// GetAddress gets the address a weavelet should listen on for a listener.
-	GetAddress(req *protos.GetAddressRequest) (*protos.GetAddressReply, error)
-
-	// ExportListener exports the given listener.
-	ExportListener(req *protos.ExportListenerRequest) (*protos.ExportListenerReply, error)
-
-	// GetRoutingInfo returns the latest routing information for the weavelet.
-	//
-	// This is a blocking method that can be processed out-of-order w.r.t.
-	// the other methods.
-	GetRoutingInfo(request *protos.GetRoutingInfo) (*protos.RoutingInfo, error)
-
-	// GetComponentsToStart is a blocking call that returns the latest set of
-	// components that should be started by the weavelet.
-	//
-	// This is a blocking method that can be processed out-of-order w.r.t.
-	// the other methods.
-	GetComponentsToStart(request *protos.GetComponentsToStart) (*protos.ComponentsToStart, error)
-
-	// RecvLogEntry enables the envelope to receive a log entry.
-	RecvLogEntry(entry *protos.LogEntry)
-
-	// RecvTraceSpans enables the envelope to receive a sequence of trace spans.
-	RecvTraceSpans(spans []trace.ReadOnlySpan) error
-}
-
-// Ensure that EnvelopeHandler remains in-sync with conn.EnvelopeHandler.
-var (
-	_ EnvelopeHandler      = conn.EnvelopeHandler(nil)
-	_ conn.EnvelopeHandler = EnvelopeHandler(nil)
-)
+type EnvelopeHandler conn.EnvelopeHandler
```

Comment:
I think the reason we copied the interface here is to avoid exposing `conn.EnvelopeHandler` in the public API since it's an internal package. And doing the opposite, putting `type EnvelopeHandler = envelope.EnvelopeHandler` in the `conn` package didn't work because of cyclic dependencies. @spetrovic77 what do you think?

Reply:
I thought that might be the reason. I started to explore another approach, moving `EnvelopeConn` into the `runtime/envelope` package. That would address the problem because both consumers of this interface would be in the same package. That can be seen here:  https://github.com/dnephin/weaver/compare/envelope-handler-interface...dnephin:weaver:envelope-handler-interface-2

With a bit more work I think it would be possible to unexport `EnvelopeConn` and solve the `weavertest` case by accepting the pipes to `Envelope`. I realized that would almost reserve the other change I had made here to remove `GetEnvelopeConn`, so I stopped that approach, but it's another option to consider.

I can revert this part of the change if it's not a good time to try and address the duplicate interfaces.

Reply:
Yeah, maybe let's revert for now so we can get the other awesome changes in this PR merged. With the type assertions below, I feel like the duplication is ugly but at least not bug-prone. 

```go
var (
	_ EnvelopeHandler      = conn.EnvelopeHandler(nil)
	_ conn.EnvelopeHandler = EnvelopeHandler(nil)
)
```

Reply:
Sounds good! I've removed the commit with this change, rebased on latest main, and squashed some of the intermediate commits.